.github/scripts/run_nm.py

```diff
@@ -0,0 +1,172 @@
+            "--reverse-sort",
+            "--demangle",
+            "--format=bsd",
+        ]
+        output = Parser.run_nm(
+            self.elf,
+            args,
+            nm=self.toolchain_prefix + "nm" if self.toolchain_prefix else "nm",
+        )
+        lines = output.splitlines()
+        symbols = []
+        symbol_pattern = re.compile(
+            r"(?P<addr>[0-9a-fA-F]+)\s+(?P<size>[0-9a-fA-F]+)\s+(?P<type>\w)\s+(?P<name>.+)"
+        )
+
+        def parse_line(line: str) -> Optional[Symbol]:
+
+            match = symbol_pattern.match(line)
+            if match:
+                addr = int(match.group("addr"), 16)
+                size = int(match.group("size"), 16)
+                type_ = match.group("type").strip().strip("\n")
+                name = match.group("name").strip().strip("\n")
+                return Symbol(name=name, addr=addr, size=size, symbol_type=type_)
+            return None
+
+        for line in lines:
+            symbol = parse_line(line)
+            if symbol:
+                symbols.append(symbol)
+
+        assert len(symbols) > 0, "No symbols found in nm output"
+        if len(symbols) != len(lines):
+            print(
+                "** Warning: Not all lines were parsed, check the output of nm. Parsed {len(symbols)} lines, given {len(lines)}"
+            )
+        if any(symbol.size == 0 for symbol in symbols):
+            print("** Warning: Some symbols have zero size, check the output of nm.")
+
+        # TODO: Populate the section and module fields from the linker map if available (-Wl,-Map=linker.map)
+        return {symbol.name: symbol for symbol in symbols}
+
+    def print(self):
+        print(f"Elf: {self.elf}")
+
+        def print_table(filter=None, filter_name=None):
+            print("\nAddress\t\tSize\tType\tName")
+            # Apply filter and sort symbols
+            symbols_to_print = {
+                name: sym
+                for name, sym in self.symbols.items()
+                if not filter or filter(sym)
+            }
+            sorted_symbols = sorted(
+                symbols_to_print.items(), key=lambda x: x[1].size, reverse=True
+            )
+
+            # Print symbols and calculate total size
+            size_total = 0
+            for name, sym in sorted_symbols:
+                print(f"{hex(sym.addr)}\t\t{sym.size}\t{sym.symbol_type}\t{sym.name}")
+                size_total += sym.size
+
+            # Print summary
+            symbol_percent = len(symbols_to_print) / len(self.symbols) * 100
+            print("-----")
+            print(f"> Total bytes: {size_total}")
+            print(
+                f"Counted: {len(symbols_to_print)}/{len(self.symbols)}, {symbol_percent:0.2f}% (filter: '{filter_name}')"
+            )
+            print("=====\n")
+
+        # Print tables with different filters
+        if (
+            self.filter is None
+            or self.filter not in ["all", "executorch", "executorch_text"]
```

Comment:
nit: isn't the None check redundant?

Reply:
you could make this easier to maintain and extend by creating a FILTER_NAME_TO_FILTER_AND_LABEL table:

```
def is_executorch_symbol(s):
    return "executorch" in s.name or s.name.startswith("et")

FILTER_NAME_TO_FILTER_AND_LABEL = {
    "all": (None, "All"),
    "executorch": (is_executorch_symbol, "ExecuTorch"),
    "executorch_text": (lambda s: is_executorch_symbol(s) and s.symbol_type.lower() == "t", "ExecuTorch .text")
}
```

then access it as
```
filter, label = FILTER_NAME_TO_FILTER_AND_LABEL.get(self.filter, FILTER_NAME_TO_FILTER_AND_LABEL["all"])
print_table(filter, label)
```

and use FILTER_NAME_TO_FILTER_AND_LABEL.keys() in the argument definition below